### PR TITLE
chore(*) migrate deprecated Envoy config to support newest version of Envoy

### PR DESCRIPTION
**Pull request by jakubdyszkiewicz**
Using `tls_context` that was deprecated in Envoy 1.12.0 https://www.envoyproxy.io/docs/envoy/latest/intro/deprecated instead of `transport_socket` prevents us to upgrade to the newest version of Envoy.

I removed a bunch of tests that checks the generated config. There is no point of checking that many cases since most advanced case covers all other cases. Those tests are a pain to maintain I suspect in the future we will get rid more of them, but for now I removed only those that in my opinion do not bring any value.

I also found that `docker/build` is not rebuilding Linux binaries therefore if you test it locally you can end up with testing old snapshot of the code.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -438,25 +438,25 @@ build/kuma-tcp-echo/linux-amd64:
 
 docker/build: docker/build/kuma-cp docker/build/kuma-dp docker/build/kumactl docker/build/kuma-injector docker/build/kuma-init docker/build/kuma-prometheus-sd docker/build/kuma-tcp-echo
 
-docker/build/kuma-cp: build/artifacts-linux-amd64/kuma-cp/kuma-cp
+docker/build/kuma-cp: build/kuma-cp/linux-amd64
```
Comment (lobkovilya) [inline Makefile:441]: Why these renames required? I believe  `build/artifacts-linux-amd64/kuma-cp/kuma-cp` should be consistent with filesystem

File: pkg/xds/envoy/clusters/clusters.go [hunk 2]
```diff
@@ -48,8 +48,20 @@ func CreateEdsCluster(ctx xds_context.Context, clusterName string, metadata *cor
...
-		TlsContext: tlsContext,
-	}), nil
+	})
+	if tlsContext != nil {
+		pbst, err := ptypes.MarshalAny(tlsContext)
+		if err != nil {
+			return nil, err
+		}
+		cluster.TransportSocket = &envoy_core.TransportSocket{
+			Name: "envoy.transport_sockets.tls",
```
Comment (lobkovilya) [inline pkg/xds/envoy/clusters/clusters.go:58]: I guess we also should update `go-control-plane` library and use `envoy_wellknown.TransportSocketTls` constant for it.

File: Makefile (hunk 1 above)
Comment (jakubdyszkiewicz) [reply Makefile]: Thanks, you are right!
I reverted the change and added comments so I don't make the same mistake again.

File: pkg/xds/envoy/clusters/clusters.go (hunk 2 above)
Comment (jakubdyszkiewicz) [reply pkg/xds/envoy/clusters/clusters.go:58]: Good point, we should update to the newest version of go-control-plane, but I'd rather have this in the separate PR. We might get into some issues and this PR is big already.
Let me create a card for this.

**Pull request merged**